public/tier0/platform.h

```diff
@@ -650,8 +657,14 @@ typedef unsigned int		uint;
 
 #endif
 
-#define  stackfree( _p )			0
-
+#define stackalloc_aligned( _size, _align )		(void*)( ( ((uintp)alloca( ALIGN_VALUE( ( _size ) + (_align ),  ( _align ) ) )) + ( _align ) ) & ~( _align - 1 ) )
+
+// We should probably always just align to 16 bytes, stackalloc just causes too many problems without this behavior. Source2 does it already.
+// #define stackalloc( _size )							stackalloc_aligned( _size, 16 )
+
+#define  stackfree( _p )			0
+// two-argument ( type, #elements) stackalloc
+#define StackAlloc( typ, nelements ) ( ( typ * )	stackalloc_aligned( ( nelements ) * sizeof(typ), 16 ) )

```

Comment:
I'm interested in knowing how did you come up with these or what is required these to be setup like that? Specifically want to hear where did you take the comment from.

Reply:
This code was taken from the leaked csgo source with one change (~_align to ~( _align - 1 ) in first line).
It is consistent with what is present in the cs2 assembly code. You can check this by compiling your program and checking the assembly code with what is in the cs2 code.

![cs2-](https://github.com/alliedmodders/hl2sdk/assets/63844820/08135d35-536c-479e-b940-bbabe177100a)

Reply:
We don't support adding the leaked content to our codebase, can you please remove all the instances of the leaked code from the pr so we can move further with it?